core/src/test/scala/org/apache/spark/benchmark/BenchmarkBase.scala

```diff
@@ -51,7 +51,7 @@ abstract class BenchmarkBase {
     val regenerateBenchmarkFiles: Boolean = System.getenv("SPARK_GENERATE_BENCHMARK_FILES") == "1"
     if (regenerateBenchmarkFiles) {
       val version = System.getProperty("java.version").split("\\D+")(0).toInt
-      val jdkString = if (version > 8) s"-jdk$version" else ""
+      val jdkString = if (version > 17) s"-jdk$version" else ""
```

Comment:
If it is using jdk8, it will be ""?

Reply:
We has been using `maven-enforcer-plugin` for the java version. We can assume Java 17+ dev environment.
```
                  <requireJavaVersion>
                    <version>${java.version}</version>
                  </requireJavaVersion>
```

Reply:
For example, we didn't consider Java 7 unto now.

Reply:
Hmm, I verified Java 8, it seems to fail in the other part.
```
$ SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "core/Test/runMain org.apache.spark.serializer.KryoBenchmark"
...
[error] '17' is not a valid choice for '-target'
[error] bad option: '-target:17'
[error] (tags / Compile / compileIncremental) Compilation failed
[error] Total time: 39 s, completed Sep 22, 2023 10:03:31 AM
```

Reply:
Hmm, so currently we cannot use jdk8 with master where Java17 is enforced?

Reply:
Yes in both ways,
- Maven build will prevent it explicitly.
- SBT also seems to hit compilation failure due to `-target:17`.